README.md

```diff
@@ -431,6 +431,97 @@ Certain WebVR `Gamepad` objects have some components with an inverted `yAxis` ca
+
+```js
+let xrGamepads = {};
+xrSession.addEventListener('inputsourceschange', onInputSourcesChange);
+
+function onInputSourcesChange(event) {
+  event.added.forEach((inputSource) => {
+    if (inputSource.gamepad) {
+      let mapping = XRGamepad.getMapping(gamepad.id, XRMappingConstants.MappingType.WebXR);
+      let xrGamepad = new XRGamepad(gamepad, mapping, inputSource.handedness);
+      xrGamepads[inputSource] = xrGamepad;
+    }
+  }
+
+  event.removed.forEach((inputSource) => {
+    if (xrGamepads[inputSource]) {
+      delete xrGamepads[inputSource];
+    }
+  });
+}
+```
+
+### WebVR Connection and Disconnection
+This sample code shows creating an `XRGamepad` for all newly connected `Gamepad` objects associated with WebVR.  When an associated `Gamepad` is disconnected the `XRGamepad` is released.
+
+```js
+let xrGamepads = {};
+navigator.addEventListener('gamepadconnected', onGamepadConnected);
+navigator.addEventListener('gamepaddisconnected', onGamepadDisconnected);
+
+function onGamepadConnected(event) {
+  let gamepad = event.gamepad;
+  if (gamepad.deviceId) {
+    let mapping = XRGamepad.getMapping(gamepad.id, XRMappingConstants.MappingType.WebVR);
+    let xrGamepad = new XRGamepad(gamepad, mapping, gamepad.handedness);
+    xrGamepads[gamepad] = xrGamepad;
+  }
+}
+
+function onGamepadDisconnected(event) {
+  if (xrGamepads[event.gamepad]) {
+      delete xrGamepads[event.gamepad];
+    }
+  }
+}
+```
+
+## Engine Loop
+On each frame, the gamepad data must be requeried and the engine must respond accordingly to the new data.  This will be the same for both WebXR and WebVR
+
+```js
+function updateGamepads() {
+  Object.key(xrGamepads).forEach((key) => {
+    let xrGamepad = xrGamepads[key];
+    xrGamepads.components.forEach((xrComponent) => {
+      const xrComponentData = xrComponent.getData();
+      MyEngine.respondToInput(key, xrComponentData);
+
+      const weightedVisualizations = xrComponent.getWeightedVisualizations();
+      MyEngine.updateInputVisuals(key, weightedVisualizations);
+    });
+  });
+}
+```
+
+> TODO Add an explanation for how to updated the visuals based on the weighted visualizations
+
+## Target Ray Origin
+In WebXR and WebVR a target ray may be drawn for motion controllers, however the origin of this ray is retrieved differently for each.
+
+### WebXR Target Ray Origin
+In WebXR, the `XRInputSource.targetRaySpace` should be passed into `XRFrame.getPose()` to determined the pointing origin.
+
+```js
+const targetRayOrigin = xrFrame.getPose(xrInputSource.targetRaySpace, xrReferenceSpace);
+```
+
+### WebVR Target Ray Origin
+In WebVR there is no implicit mechanism for retrieving a target ray origin.  Instead, it must be retrieved from the the mapping via the `XRGamepad`
+```js
+const targetRayOrigin = xrGamepad.targetRayOrigin;
```

Comment:
Hm... maybe I just misread this previously: I thought that `targetRayOrigin` was a static offset from the base gamepad pose? That would appear to not be the case here if this is meant to be ~equivalent to the WebXR code above.

Reply:
Gah.  No that's me moving too fast.  I'll get an update pushed in a moment.